src/build-data/buildh.in

```diff
@@ -210,20 +210,26 @@
 #define BOTAN_SYSTEM_RNG_POLL_TIMEOUT_MS 20
 
 /*
-How many times to read from the RDRAND/RDSEED RNGs.
-Each read generates 32 bits of output
+* How many bytes of output to read from RDRAND for seeding purposes.
+* We default to 8192 bytes which is (according to Intel documentation)
+* guaranteed to cause at least one reseeding of the internal RNG.
 */
-#define BOTAN_ENTROPY_INTEL_RNG_POLLS 32
+#define BOTAN_ENTROPY_INTEL_RDRAND_BYTES 8192
 
 /*
-According to Intel, RDRAND is guaranteed to generate a random
-number within 10 retries on a working CPU
+* How many bytes of output to read from RDSEED for seeding purposes.
+*/
+#define BOTAN_ENTROPY_INTEL_RDSEED_BYTES 128
+
+/*
+* According to Intel, RDRAND is guaranteed to generate a random
+* number within 10 retries on a working CPU
 */
 #define BOTAN_ENTROPY_RDRAND_RETRIES 10
 
 /*
-* RdSeed is not guaranteed to generate a random number within a specific number of retries
-* Define the number of retries here
+* RdSeed is not guaranteed to generate a random number within a
+* specific number of retries. Define the number of retries here.
 */
 #define BOTAN_ENTROPY_RDSEED_RETRIES 20
```

Comment:
similar discussion here. From the Intel docs:

> 5.3.1.1 Synchronous applications
If the application is not latency-sensitive, then it can simply retry the RDSEED
instruction indefinitely, though it is recommended that a PAUSE instruction be placed in
the retry loop. In the worst-case scenario, where multiple threads are invoking RDSEED
continually, the delays can be long, but the longer the delay, the more likely (with an
exponentially increasing probability) that the instruction will return a result.
If the application is latency-sensitive, then applications should either sleep or fall back to
generating seed values from RDRAND.

> 5.3.1.2 Asynchronous applications
The application should be prepared to give up on RDSEED after a small number of
retries, where "small" is somewhere between 1 and 100, depending on the application's
sensitivity to delays. As with synchronous applications, it is recommended that a PAUSE
instruction be inserted into the retry loop.
Applications needing a more aggressive approach can alternate between RDSEED and
RDRAND, pulling seeds from RDSEED as they are available and filling a RDRAND
buffer for future 512:1 reduction when they are not.

Mhh okay. Maybe the most secure should be our default: indefinitely loop with PAUSE. But this should be configurable.

Reply:
I also did some experimentation here, while running one or more instances of a program that is just `for(;;) { rdseed64_step(&output); }`

With all cores pinned in RDSEED, the maximum number of retry attempts I ever saw was 230. So I set an upper limit of 512 attempts - if we exceed that limit, we stop the poll. I don't want to loop forever because we could run into an issue where the firmware is buggy and RDSEED never succeeds.

Including PAUSE didn't seem to make any difference in the success rate that I could tell, but since Intel recommends it and it doesn't seem to cause problems I added it.